stanford_capx.entity.hooks.inc

```diff
@@ -253,19 +253,50 @@ function stanford_capx_entity_delete($entity, $type) {
 function stanford_capx_entity_load($entities, $type) {
 
   $ids = array_keys($entities);
-  $meta = &drupal_static(__FUNCTION__, NULL);
+  $meta = &drupal_static(__FUNCTION__, array());
 
-  if (empty($meta)) {
-    $meta = db_select("capx_profiles", 'capx')
+  if (!isset($meta[$type])) {
+    $meta[$type] = array();
+  }
+
+  // Some sites have wwaaayyy too many items in the capx_profiles table for this
+  // to load with an appropriate amount of memory. You can thank Zach Chandler
+  // for finding out this edge case. As such, we will apply two different
+  // stategies for fetching the CAPx metadata. If the amount of profile entries
+  // are over a value then switch from one big cached call to multiple pings of
+  // the database. We are trading PHP memory for hammering the DB with multiple
+  // queries.
+
+  $count = db_select('capx_profiles')
+    ->fields(NULL, array('id'))
+    ->countQuery()
+    ->execute()
+    ->fetchField();
+
+  if ($count > variable_get('stanford_capx_entity_load_limit', 10000)) {
+    $query_entities = array_diff_key($entities, $meta[$type]);
+    $query_results = db_select("capx_profiles", 'capx')
       ->fields('capx')
+      ->condition('entity_type', $type)
+      ->condition('entity_id', array_keys($query_entities))
       ->execute()
       ->fetchAllAssoc('entity_id');
+
+    $meta[$type] = $meta[$type] + $query_results;
+  }
+  else {
+    if (empty($meta[$type])) {
+      $meta[$type] = db_select("capx_profiles", 'capx')
+        ->fields('capx')
+        ->execute()
+        ->fetchAllAssoc('entity_id');
+    }
   }
 
   // Add the capx information if it matches.
   foreach ($ids as $entity_id) {
     // If we do not find a match then carry on.
-    if (isset($meta[$entity_id]) && $meta[$entity_id]->entity_type == $type) {
+    if (isset($meta[$type][$entity_id]) && $meta[$type][$entity_id]->entity_type == $type) {
```

Comment:
Adding the $type key will avoid having that clash of entity ids.  We probably haven't run in to this issue yet because most are just importing in to nodes.